lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -952,4 +977,176 @@ void SessionStateEnforcer::update_session_with_policy(
                              get_upf_n3_addr(), session->get_upf_local_teid());
 }
 
+void SessionStateEnforcer::init_policy_reauth(const std::string& imsi,
+                                              SessionMap& session_map,
+                                              PolicyReAuthRequest request,
+                                              PolicyReAuthAnswer& answer_out,
+                                              SessionUpdate& session_update) {
+  const std::string& session_id = request.session_id();
+  auto it = session_map.find(imsi);
+  if (it == session_map.end()) {
+    MLOG(MERROR) << "Could not find subscriber " << imsi
+                 << " during policy ReAuth";
+    answer_out.set_result(ReAuthResult::SESSION_NOT_FOUND);
+    return;
+  }
+  // For empty session_id, apply changes to all sessions of subscriber
+  // Changes are applied on a best-effort basis, so failures for one session
+  // won't stop changes from being applied for subsequent sessions.
+  if (session_id == "") {
+    for (const auto& session : it->second) {
+      init_policy_reauth_for_session(imsi, request, session, session_update);
+    }
+    answer_out.set_result(ReAuthResult::UPDATE_INITIATED);
+    return;
+  }
+
+  SessionSearchCriteria criteria(imsi, IMSI_AND_SESSION_ID, session_id);
+  auto session_it = session_store_.find_session(session_map, criteria);
+  if (!session_it) {
+    MLOG(MERROR) << "Could not find session " << session_id
+                 << " during policy ReAuth";
+    answer_out.set_result(ReAuthResult::SESSION_NOT_FOUND);
+    return;
+  }
+  auto& session = **session_it;

```

Comment:
Need to check the state of session before policy_reauth Init.  If session is IDLE state then modification will not be considered. It should return as UPDATE_NOT_NEEDED or OTHER_FAILURE.

Reply:
Yeah, Added code for return `OTHER_FAILURE` if session is not an `ACTIVE` state as suggested.